rootex/framework/entity_factory.cpp

```diff
@@ -42,8 +42,18 @@ EntityID EntityFactory::s_CurrentEditorID = -ROOT_ENTITY_ID;
 void EntityFactory::RegisterAPI(sol::table& rootex)
 {
 	sol::usertype<EntityFactory> entityFactory = rootex.new_usertype<EntityFactory>("EntityFactory");
-	entityFactory["Create"] = [](TextResourceFile* t) { return EntityFactory::GetSingleton()->createEntity(t); };
-	entityFactory["CreateFromClass"] = [](TextResourceFile* t) { return EntityFactory::GetSingleton()->createEntityFromClass(t); };
+	entityFactory["Create"] = [](TextResourceFile* t) {
+		Ref<Entity> newEntity =  EntityFactory::GetSingleton()->createEntity(t); 
+		newEntity->evaluateScriptOverrides();
+		newEntity->call("onBegin", { newEntity });
+		return newEntity;
+	};
+	entityFactory["CreateFromClass"] = [](TextResourceFile* t) { 
+		Ref<Entity> newEntity = EntityFactory::GetSingleton()->createEntityFromClass(t); 
+		newEntity->evaluateScriptOverrides();
+		newEntity->call("onBegin", { newEntity });
+		return newEntity;
+	};
```

Comment:
Reminder to remove the `onBegin()` calls when we add a constructor function to the scripts